crates/txtx-core/src/validation/hcl_validator/block_processors.rs

```diff
@@ -0,0 +1,315 @@
+    ValidationError,
+};
+
+/// Extract position from a block's identifier span
+///
+/// Converts the block's identifier span to a Position using the source mapper.
+/// Returns a default position (1, 1) if the block has no span information.
+///
+/// This helper consolidates the repeated pattern of extracting positions
+/// from block identifiers, reducing code duplication across processor functions.
+fn extract_block_position(block: &Block, source_mapper: &SourceMapper) -> Position {
+    block.ident.span()
+        .as_ref()
+        .map(|span| {
+            let (line, col) = source_mapper.span_to_position(span);
+            Position::new(line, col)
+        })
+        .unwrap_or_default()
+}
+
+/// Process a block during the collection phase.
+pub fn process_block(
+    block: &Block,
+    block_type: BlockType,
+    addon_specs: &HashMap<String, Vec<(String, CommandSpecification)>>,
+    source_mapper: &SourceMapper,
+) -> Result<Vec<CollectedItem>, ValidationError> {
+    match block_type {
+        BlockType::Signer => process_signer(block),
+        BlockType::Variable => process_variable(block, source_mapper),
+        BlockType::Output => process_output(block),
+        BlockType::Secret => process_secret(block),
+        BlockType::Action => process_action(block, addon_specs, source_mapper),
+        BlockType::Flow => process_flow(block, source_mapper),
+        BlockType::Addon | BlockType::Unknown => Ok(Vec::new()),
+    }
+}
+
+fn process_signer(block: &Block) -> Result<Vec<CollectedItem>, ValidationError> {
+    let name = block.labels.extract_name()
+        .ok_or(ValidationError::MissingLabel("signer name"))?;
+
+    let signer_type = block.labels.extract_type()
+        .ok_or(ValidationError::MissingLabel("signer type"))?;
+
+    Ok(vec![
+        CollectedItem::Definition(DefinitionItem::Signer {
+            name: name.to_string(),
+            signer_type: signer_type.to_string(),
+        })
+    ])
+}
+
+fn process_variable(block: &Block, source_mapper: &SourceMapper) -> Result<Vec<CollectedItem>, ValidationError> {
+    use txtx_addon_kit::hcl::visit::{visit_expr, Visit};
+    use txtx_addon_kit::hcl::expr::{Expression, TraversalOperator};
+
+    let name = block.labels.extract_name()
+        .ok_or(ValidationError::MissingLabel("variable name"))?;
+
+    let position = extract_block_position(block, source_mapper);
+
+    // Extract dependencies from the variable's value
+    let mut dependencies = Vec::new();
+
+    struct DependencyExtractor<'a> {
+        dependencies: &'a mut Vec<String>,
+    }
+
+    impl<'a> Visit for DependencyExtractor<'a> {
+        fn visit_expr(&mut self, expr: &Expression) {
+            // Use pattern matching to extract variable dependencies
+            if let Expression::Traversal(traversal) = expr {
+                traversal.expr.as_variable()
+                    .filter(|name| matches!(name.as_str(), "var" | "variable"))
```

Comment:
Currently "var" is not supported.

I think we'd be open to it, but this would require other changes so let's remove and add an issue if you want this.